nautobot/ipam/api/serializers.py

```diff
@@ -206,37 +141,18 @@ def validate(self, data):
 #
 
 
-class PrefixSerializer(
-    NautobotModelSerializer, TaggedModelSerializerMixin, StatusModelSerializerMixin, RoleModelSerializerMixin
-):
+class PrefixSerializer(NautobotModelSerializer, TaggedModelSerializerMixin):
     url = serializers.HyperlinkedIdentityField(view_name="ipam-api:prefix-detail")
     family = ChoiceField(choices=IPAddressFamilyChoices, read_only=True)
     prefix = IPFieldSerializer()
     type = ChoiceField(choices=PrefixTypeChoices, default=PrefixTypeChoices.TYPE_NETWORK)
-    location = NestedLocationSerializer(required=False, allow_null=True)
-    vrf = NestedVRFSerializer(required=False, allow_null=True)
-    tenant = NestedTenantSerializer(required=False, allow_null=True)
-    vlan = NestedVLANSerializer(required=False, allow_null=True)
-    rir = NestedRIRSerializer(required=False, allow_null=True)
 
     class Meta:
         model = Prefix
-        fields = [
-            "url",
-            "family",
-            "prefix",
-            "type",
-            "location",
-            "vrf",
-            "tenant",
-            "vlan",
-            "status",
-            "role",
-            "rir",
-            "date_allocated",
-            "description",
-        ]
         read_only_fields = ["family"]
+        # excluded here so that the user do not need to pass in a `prefix_length` value
+        # when doing a POST request.
+        exclude = ["prefix_length"]
```

Comment:
shouldn't it be in `read_only_fields` rather than excluded entirely?

Reply:
Yes, make it read-only for now. We'll revisit it in the Namespace work.